sw/otbn/crypto/rsa_keygen.s

```diff
@@ -135,14 +227,64 @@ derive_d:
   la       x12, tmp_scratchpad
   la       x13, rsa_d
   la       x14, rsa_cofactor
-  la       x15, rsa_pq
+  la       x15, rsa_d_crt
   jal      x1, modinv_f4
 
   /* Reset the limb count.
        x30 <= (plen*2) >> 1 = n */
   srli     x30, x30, 1
   ret
 
+/**
+ * Derive a private RSA exponent CRT component from the private exponent d.
+ *
+ * The supplied value at `dptr_pm1` must be n limbs as a requirement of `div`,
+ * but this function ignores the most significant n/2 limbs, zero-extending the
+ * least significant n/2 limbs to the full buffer.
```

Comment:
I think `div` could be modified to not require the arguments to be the same length. A standalone Montgomery reduction might _not_ be faster overall here, because the constants for `p-1` and `q-1` are not otherwise needed. But perhaps we could save some time and more importantly (given that RSA keygen time is dominated by primality testing anyway) space/complexity by adjusting `div` to be a little more flexible. Again, shouldn't block the PR, more of a future-improvements thought.